src/components/Launch/LaunchWorkflowForm/BlobInput.tsx

```diff
@@ -0,0 +1,129 @@
+import {
+    FormControl,
+    FormHelperText,
+    InputLabel,
+    MenuItem,
+    Select,
+    TextField,
+    Typography
+} from '@material-ui/core';
+import { makeStyles, Theme } from '@material-ui/core/styles';
+import { isObject } from 'lodash';
+import { BlobDimensionality } from 'models';
+import * as React from 'react';
+import {
+    blobFormatHelperText,
+    blobUriHelperText,
+    defaultBlobValue
+} from './constants';
+import { BlobValue, InputProps } from './types';
+import { getLaunchInputId } from './utils';
+
+const useStyles = makeStyles((theme: Theme) => ({
+    dimensionalityInput: {
+        flex: '1 1 auto',
+        marginLeft: theme.spacing(2)
+    },
+    formatInput: {
+        flex: '1 1 auto'
+    },
+    inputContainer: {
+        borderLeft: `1px solid ${theme.palette.divider}`,
+        marginTop: theme.spacing(1),
+        paddingLeft: theme.spacing(1)
+    },
+    metadataContainer: {
+        display: 'flex',
+        marginTop: theme.spacing(1),
+        width: '100%'
+    }
+}));
+
+/** A micro form for entering the values related to a Blob Literal */
+export const BlobInput: React.FC<InputProps> = props => {
+    const styles = useStyles();
+    const { error, label, name, onChange, value: propValue } = props;
+    const blobValue = isObject(propValue)
+        ? (propValue as BlobValue)
+        : defaultBlobValue;
+    const hasError = !!error;
+    const helperText = hasError ? error : props.helperText;
+
+    const handleUriChange = ({
+        target: { value: uri }
+    }: React.ChangeEvent<HTMLInputElement>) => {
+        onChange({
+            ...blobValue,
+            uri
+        });
+    };
+
+    const handleFormatChange = ({
+        target: { value: format }
+    }: React.ChangeEvent<HTMLInputElement>) => {
+        onChange({
+            ...blobValue,
+            format
+        });
+    };
+
+    const handleDimensionalityChange = ({
+        target: { value }
+    }: React.ChangeEvent<{ value: unknown }>) => {
+        onChange({
+            ...blobValue,
+            dimensionality: value as BlobDimensionality
```

Comment:
Similar suggestion for type guards here if possible

Reply:
I'm not really sure what action we would take here upon checking the value and seeing that it's _not_ a dimensionality. It's coming from a `Select` component, not a text field. So it's not possible for it to be anything other than the values we provide to that component (which are type-checked elsewhere). This cast is because the type of the change event doesn't allow for specifying the type of the value.